src/executor/stack/executor.rs

```diff
@@ -248,6 +271,7 @@ pub trait PrecompileSet {
 	/// If the provided address is not a precompile, returns None.
 	fn execute(
 		&self,
+		handle: &mut impl PrecompileHandle,
 		address: H160,
 		input: &[u8],
```

Comment:
You can move all parameters other than `address` to the handle trait. The lifetime should work, as everything will be within the scope of the overall handle lifetime.

Reply:
Sounds good. While updating our precompiles I already noticed `target_gas` was not very useful anymore since we can obtain the remaining gas.

Reply:
Hmm also moved the address, but that shouldn't make a difference.